generated/language/src/test/java/com/google/cloud/language/v1/spring/LanguageAutoConfigurationTests.java

```diff
@@ -0,0 +1,310 @@
+import com.google.api.gax.core.InstantiatingExecutorProvider;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.TransportChannel;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.LanguageServiceSettings;
+import com.google.cloud.spring.autoconfigure.core.GcpContextAutoConfiguration;
+import java.io.IOException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.threeten.bp.Duration;
+
+@ExtendWith(MockitoExtension.class)
+class LanguageAutoConfigurationTests {
+
+  private static final String SERVICE_CREDENTIAL_LOCATION =
+      "src/test/resources/fake-credential-key.json";
+  private static final String SERVICE_CREDENTIAL_CLIENT_ID = "45678";
+  private static final String SERVICE_OVERRIDE_CLIENT_ID = "56789";
+
+  @Mock private TransportChannel mockTransportChannel;
+  @Mock private ApiCallContext mockApiCallContext;
+  @Mock private TransportChannelProvider mockTransportChannelProvider;
+  @Mock private CredentialsProvider mockCredentialsProvider;
+
+  private ApplicationContextRunner contextRunner =
+      new ApplicationContextRunner()
+          .withConfiguration(
+              AutoConfigurations.of(
+                  GcpContextAutoConfiguration.class, LanguageServiceSpringAutoConfiguration.class));
+
+  @Test
+  void testLanguageServiceClientCreated() {
+    this.contextRunner.run(
+        ctx -> {
+          LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+          assertThat(client).isNotNull();
+        });
+  }
+
+  @Test
+  void testServiceCredentialsFromProperties() {
+    this.contextRunner
+        .withPropertyValues(
+            "com.google.cloud.language.v1.spring.auto.language-service.credentials.location=file:"
+                + SERVICE_CREDENTIAL_LOCATION)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              Credentials credentials =
+                  client.getSettings().getCredentialsProvider().getCredentials();
+              assertThat(((ServiceAccountCredentials) credentials).getClientId())
+                  .isEqualTo(SERVICE_CREDENTIAL_CLIENT_ID);
+            });
+  }
+
+  @Test
+  void testShouldUseCredentialsBeanWhenPropertiesNotProvided() {
+    this.contextRunner.run(
+        ctx -> {
+          LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+          assertThat(client.getSettings().getCredentialsProvider()).isNotNull();
```

Comment:
Wouldn't this be true even when properties were provided, like in the test above?